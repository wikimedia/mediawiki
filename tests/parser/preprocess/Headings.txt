These should become headings:

== h ==<!--c1-->

== h == 	<!--c1-->

== h ==<!--c1--> 	

== h == 	<!--c1--> 	

== h ==<!--c1--><!--c2-->

== h == 	<!--c1--><!--c2-->

== h ==<!--c1--><!--c2--> 	

== h == 	<!--c1--><!--c2--> 	

== h == 	<!--c1-->  <!--c2-->

== h ==<!--c1-->  <!--c2--> 	

== h == 	<!--c1-->  <!--c2--> 	

== h ==<!--c1--><!--c2--><!--c3-->

== h ==<!--c1-->  <!--c2--><!--c3-->

== h ==<!--c1--><!--c2-->  <!--c3-->

== h ==<!--c1-->  <!--c2-->  <!--c3-->

== h ==  <!--c1--><!--c2--><!--c3-->

== h ==  <!--c1-->  <!--c2--><!--c3-->

== h ==  <!--c1--><!--c2-->  <!--c3-->

== h ==  <!--c1-->  <!--c2-->  <!--c3-->

== h ==<!--c1--><!--c2--><!--c3-->  

== h ==<!--c1-->  <!--c2--><!--c3-->  

== h ==<!--c1--><!--c2-->  <!--c3-->  

== h ==<!--c1-->  <!--c2-->  <!--c3-->  

== h ==  <!--c1--><!--c2--><!--c3-->  

== h ==  <!--c1-->  <!--c2--><!--c3-->  

== h ==  <!--c1--><!--c2-->  <!--c3-->  

== h ==  <!--c1-->  <!--c2-->  <!--c3-->  


Those are not working:

== h ==<!--c1--> 	<!--c2-->

== h == 	<!--c1--> 	<!--c2-->

== h ==<!--c1--> 	<!--c2--> 	

== h == x <!--c1--><!--c2--><!--c3-->  

== h ==<!--c1--> x <!--c2--><!--c3-->  

== h ==<!--c1--><!--c2--><!--c3--> x 
