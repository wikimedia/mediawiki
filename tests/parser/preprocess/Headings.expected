<root>These should become headings:

<h level="2" i="1">== h ==<comment>&lt;!--c1--&gt;</comment></h>

<h level="2" i="2">== h == 	<comment>&lt;!--c1--&gt;</comment></h>

<h level="2" i="3">== h ==<comment>&lt;!--c1--&gt;</comment> 	</h>

<h level="2" i="4">== h == 	<comment>&lt;!--c1--&gt;</comment> 	</h>

<h level="2" i="5">== h ==<comment>&lt;!--c1--&gt;</comment><comment>&lt;!--c2--&gt;</comment></h>

<h level="2" i="6">== h == 	<comment>&lt;!--c1--&gt;</comment><comment>&lt;!--c2--&gt;</comment></h>

<h level="2" i="7">== h ==<comment>&lt;!--c1--&gt;</comment><comment>&lt;!--c2--&gt;</comment> 	</h>

<h level="2" i="8">== h == 	<comment>&lt;!--c1--&gt;</comment><comment>&lt;!--c2--&gt;</comment> 	</h>

<h level="2" i="9">== h == 	<comment>&lt;!--c1--&gt;</comment>  <comment>&lt;!--c2--&gt;</comment></h>

<h level="2" i="10">== h ==<comment>&lt;!--c1--&gt;</comment>  <comment>&lt;!--c2--&gt;</comment> 	</h>

<h level="2" i="11">== h == 	<comment>&lt;!--c1--&gt;</comment>  <comment>&lt;!--c2--&gt;</comment> 	</h>

<h level="2" i="12">== h ==<comment>&lt;!--c1--&gt;</comment><comment>&lt;!--c2--&gt;</comment><comment>&lt;!--c3--&gt;</comment></h>

<h level="2" i="13">== h ==<comment>&lt;!--c1--&gt;</comment>  <comment>&lt;!--c2--&gt;</comment><comment>&lt;!--c3--&gt;</comment></h>

<h level="2" i="14">== h ==<comment>&lt;!--c1--&gt;</comment><comment>&lt;!--c2--&gt;</comment>  <comment>&lt;!--c3--&gt;</comment></h>

<h level="2" i="15">== h ==<comment>&lt;!--c1--&gt;</comment>  <comment>&lt;!--c2--&gt;</comment>  <comment>&lt;!--c3--&gt;</comment></h>

<h level="2" i="16">== h ==  <comment>&lt;!--c1--&gt;</comment><comment>&lt;!--c2--&gt;</comment><comment>&lt;!--c3--&gt;</comment></h>

<h level="2" i="17">== h ==  <comment>&lt;!--c1--&gt;</comment>  <comment>&lt;!--c2--&gt;</comment><comment>&lt;!--c3--&gt;</comment></h>

<h level="2" i="18">== h ==  <comment>&lt;!--c1--&gt;</comment><comment>&lt;!--c2--&gt;</comment>  <comment>&lt;!--c3--&gt;</comment></h>

<h level="2" i="19">== h ==  <comment>&lt;!--c1--&gt;</comment>  <comment>&lt;!--c2--&gt;</comment>  <comment>&lt;!--c3--&gt;</comment></h>

<h level="2" i="20">== h ==<comment>&lt;!--c1--&gt;</comment><comment>&lt;!--c2--&gt;</comment><comment>&lt;!--c3--&gt;</comment>  </h>

<h level="2" i="21">== h ==<comment>&lt;!--c1--&gt;</comment>  <comment>&lt;!--c2--&gt;</comment><comment>&lt;!--c3--&gt;</comment>  </h>

<h level="2" i="22">== h ==<comment>&lt;!--c1--&gt;</comment><comment>&lt;!--c2--&gt;</comment>  <comment>&lt;!--c3--&gt;</comment>  </h>

<h level="2" i="23">== h ==<comment>&lt;!--c1--&gt;</comment>  <comment>&lt;!--c2--&gt;</comment>  <comment>&lt;!--c3--&gt;</comment>  </h>

<h level="2" i="24">== h ==  <comment>&lt;!--c1--&gt;</comment><comment>&lt;!--c2--&gt;</comment><comment>&lt;!--c3--&gt;</comment>  </h>

<h level="2" i="25">== h ==  <comment>&lt;!--c1--&gt;</comment>  <comment>&lt;!--c2--&gt;</comment><comment>&lt;!--c3--&gt;</comment>  </h>

<h level="2" i="26">== h ==  <comment>&lt;!--c1--&gt;</comment><comment>&lt;!--c2--&gt;</comment>  <comment>&lt;!--c3--&gt;</comment>  </h>

<h level="2" i="27">== h ==  <comment>&lt;!--c1--&gt;</comment>  <comment>&lt;!--c2--&gt;</comment>  <comment>&lt;!--c3--&gt;</comment>  </h>


Those are not working:

== h ==<comment>&lt;!--c1--&gt;</comment> 	<comment>&lt;!--c2--&gt;</comment>

== h == 	<comment>&lt;!--c1--&gt;</comment> 	<comment>&lt;!--c2--&gt;</comment>

== h ==<comment>&lt;!--c1--&gt;</comment> 	<comment>&lt;!--c2--&gt;</comment> 	

== h == x <comment>&lt;!--c1--&gt;</comment><comment>&lt;!--c2--&gt;</comment><comment>&lt;!--c3--&gt;</comment>  

== h ==<comment>&lt;!--c1--&gt;</comment> x <comment>&lt;!--c2--&gt;</comment><comment>&lt;!--c3--&gt;</comment>  

== h ==<comment>&lt;!--c1--&gt;</comment><comment>&lt;!--c2--&gt;</comment><comment>&lt;!--c3--&gt;</comment> x 
</root>